pkg/table/tables/partition.go

```diff
@@ -1370,21 +1370,7 @@ func (t *partitionedTable) locateRangeColumnPartition(ctx expression.BuildContex
 		return 0, errors.Trace(lastError)
 	}
 	if idx >= len(upperBounds) {
-		// The data does not belong to any of the partition returns `table has no partition for value %s`.
-		var valueMsg string
-		if t.meta.Partition.Expr != "" {
-			e, err := expression.ParseSimpleExpr(ctx, t.meta.Partition.Expr, expression.WithTableInfo("", t.meta))
-			if err == nil {
-				val, _, err := e.EvalInt(ctx.GetEvalCtx(), chunk.MutRowFromDatums(r).ToRow())
-				if err == nil {
-					valueMsg = strconv.FormatInt(val, 10)
-				}
-			}
```

Comment:
Seems this branch is only for range partitioning table, so I removed it. PTAL @mjonss

Reply:
Right, if it is RANGE COLUMNS, then `t.meta.Partition.Expr` should be empty, since it should only allow a list of columns.